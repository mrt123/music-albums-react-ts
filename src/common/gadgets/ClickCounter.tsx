import { MouseEventHandler, useState } from "react";

const style = {
  borderRadius: "30%",
  border: "2px solid white",
  padding: "11px",
  fontSize: "21px",
  lineHeight: "8px",
  height: "30px",
  cursor: "pointer",
  userSelect: "none" as "none",
};

const ClickCounter: React.FC = () => {
  const [counter, setCounter] = useState<number>(0);

  const increaseCounter: MouseEventHandler = (e) => {
    setCounter(counter + 1);
  };

  return (
    <div data-testid="clickCounter" style={style} onClick={increaseCounter}>
      {counter}
    </div>
  );
};

export default ClickCounter;
