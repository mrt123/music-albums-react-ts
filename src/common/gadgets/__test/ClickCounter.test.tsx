import { fireEvent, render } from "@testing-library/react";
import exp from "constants";
import ClickCounter from "../ClickCounter";

describe("ClickCounter.tsx", () => {
  it("matches snapshot when there were no clicks", () => {
    const { container } = render(<ClickCounter />);
    expect(container).toMatchSnapshot();
  });

  it("increases from 0 to 1, after 1 click", () => {
    const { getByTestId } = render(<ClickCounter />);

    const clickCounterElement = getByTestId("clickCounter");
    expect(clickCounterElement).toHaveTextContent("0");

    fireEvent(
      clickCounterElement,

      new MouseEvent("click", {
        bubbles: true,
        cancelable: true,
      })
    );

    expect(clickCounterElement).toHaveTextContent("1");
  });

  it("renders 5 after 5 clicks", () => {
    const { getByTestId } = render(<ClickCounter />);
    const clickCounterElement = getByTestId("clickCounter");

    expect(clickCounterElement).toHaveTextContent("0");

    for (var i = 0; i < 5; i++) {
      fireEvent(
        clickCounterElement,

        new MouseEvent("click", {
          bubbles: true,
          cancelable: true,
        })
      );
    }
    expect(clickCounterElement).toHaveTextContent("5");
  });
});
