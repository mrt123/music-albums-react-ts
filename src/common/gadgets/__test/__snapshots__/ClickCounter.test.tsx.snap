// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClickCounter.tsx matches snapshot when there were no clicks 1`] = `
<div>
  <div
    data-testid="clickCounter"
    style="border-radius: 30%; border: 2px solid white; padding: 11px; font-size: 21px; line-height: 8px; height: 30px; cursor: pointer; user-select: none;"
  >
    0
  </div>
</div>
`;
